helios/views.py

```diff
@@ -1306,8 +1303,8 @@ def voters_upload(request, election):
           problems.append("your CSV file could not be processed. Please check that it is a proper CSV file.")
 
         # check if voter emails look like emails
-        if False in [validate_email(v['email']) for v in voters]:
-          problems.append("those don't look like correct email addresses. Are you sure you uploaded a file with email address as second field?")
+        if False in [v['voter_type'] in AUTH_SYSTEMS for v in voters]:
+          problems.append("those don't look like valid auth methods. Are you sure you uploaded a file with voter type as first field?")
```

Comment:
good that we're checking this here, too. My previous comment still holds, I think we want validation right at data insertion, too.

One thing that is now missing is the call to `validate_email`, which has been useful to ensure that email addresses are not bogus. I would be okay with a small break in abstraction here and a call to `validate_email` if `voter_type` is password. A more correct longer term approach would be a `validate_record` call that is specific to the auth system, but I think it's okay to hold off on that to a subsequent ticket.

I would like to ensure we don't regress here, though, and skip the email validation we had before.

Reply:
should we validate_email in the voter_file instead of here then, perhaps? right now it's only doing a spot-check as is, it's only looking at the first 5 lines :grimacing: